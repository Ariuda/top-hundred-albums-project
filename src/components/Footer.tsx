import React from 'react';
import { connect } from 'react-redux';
import { RootState } from '../store/reducers';
import dayjs from 'dayjs';

import { APIAuthor } from '../entities';

interface Props {
    APIAuthorDetails: APIAuthor
}

export const Footer: React.FC<Props> = ({ APIAuthorDetails }) => {
    const { lastUpdated, link, name, rights, title } = APIAuthorDetails;

    if(lastUpdated && link && name && rights && title) {
        return (
            <footer className="footer">
                <div className="container">
                    <p>Powered by {title}, {rights}</p>
                    <p>Last updated on {dayjs(lastUpdated).format('DD-MM-YYYY [at] HH:mm:ss')}</p>
                    <p>You can visit the {name} by <a href={link} target="_blank">clicking here</a></p>
                </div>
            </footer>
        )
    } 
    return null;
};

const mapStateToProps = (state: RootState) => {
    return {
        APIAuthorDetails: state.albumsState.APIAuthor
    }
}

export default connect(mapStateToProps)(Footer);