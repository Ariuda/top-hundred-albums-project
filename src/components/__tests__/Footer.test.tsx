import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Footer } from '../Footer';

const APIAuthorDetails = {
    name: 'name',
    link: 'link',
    rights: 'rights', 
    title: 'title', 
    lastUpdated: 'last-updated' 
}

it('should render the footer when it gets all the data from the api', () => {
    const wrapper = shallow(<Footer APIAuthorDetails={APIAuthorDetails} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should not render the footer if it does not get all the data from the api', () => {
    const emtpyAuthorDetails = {
        name: '',
        link: '',
        rights: '', 
        title: '', 
        lastUpdated: '' 
    }
    const wrapper = shallow(<Footer APIAuthorDetails={emtpyAuthorDetails} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
