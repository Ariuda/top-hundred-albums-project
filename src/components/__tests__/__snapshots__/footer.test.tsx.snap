// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render the footer if it does not get all the data from the api 1`] = `""`;

exports[`should render the footer when it gets all the data from the api 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <p>
      Powered by 
      title
      , 
      rights
    </p>
    <p>
      Last updated on 
      Invalid Date
    </p>
    <p>
      You can visit the 
      name
       by 
      <a
        href="link"
        target="_blank"
      >
        clicking here
      </a>
    </p>
  </div>
</footer>
`;
